### PR TITLE
Support Models' and SubCategories' selection in `SelectionSet`

### PR DESCRIPTION
**Pull request by grigasp**
Closes https://github.com/iTwin/itwinjs-core/issues/7441.

**From `NextVersion.md`:**

There are two similar selection-related concepts in `@itwin/core-frontend` - `SelectionSet` and `HiliteSet`. The former is generally used by interactive tools (e.g. the "Move element" tool), so it contains what tools think is selected. The latter is used by the graphics system to know what elements to highlight, so it contains what users think is selected. Generally, we want the two sets to be in sync to avoid confusion why tools act on different elements than what users think are selected. Keeping them in sync was not always possible, because `HiliteSet` may store Model and SubCategory ids, but `SelectionSet` could only store Element ids. So we could end up in situations where a Model id is added to `HiliteSet` and `SelectionSet` is empty, making users think that all elements in that model are selected, but tools not knowing anything about it.

To alleviate this problem, the `SelectionSet`-related APIs have been enhanced to support storing Model and SubCategory ids, similar to what `HiliteSet` does. The change has been made in a backwards compatible way, so all existing code using `SelectionSet` should continue to work as before:

- `SelectionSet` modification methods `add`, `addAndRemove`, `remove`, `replace` now, in addition to existing `Id64Arg` argument, accept the `SelectableIds` structure.
- `SelectionSetEvent` attributes `added` and `removed` have been deprecated, but continue to work as before, containing only element ids. In addition, the event object now contains new `additions` and `removals` attributes, which are instances of `SelectableIds` and contain all ids that were added or removed from the selection set, including those of Model and SubCategory.

Because the `SelectionSet` now stores additional types of ids, existing code that listens to `onChange` event may start getting extra invocations that don't affect the element selection (e.g. `SelectAddEvent` with `added: []` and `additions: { models: ["0x1"] }`). Also, the `isActive` getter may return `true` even though `elements` set is empty.

### CODE REVIEW

File: common/api/core-frontend.api.md [hunk 1]
```diff
@@ -10654,9 +10658,21 @@ export class SectionDrawingModelState extends DrawingModelState {
     static get className(): string;
 }
 
+// @public
+export interface SelectableIds {
```
Comment (pmconne) [inline common/api/core-frontend.api.md:10662]: Why make every property required here if you're going to use `Partial<SelectableIds>` everywhere in the API?

Comment (grigasp) [reply common/api/core-frontend.api.md:10662]: Agree, makes no sense. Made the members optional.

**Review by pmconne — APPROVED** [review]

File: core/frontend/src/SelectionSet.ts [hunk 2]
```diff
@@ -209,233 +176,559 @@ export class HiliteSet {
...
+
+  /** The IDs of the selected subcategories.
+   * @note Do not modify this set directly. Instead, use methods like [[SelectionSet.add]].
+   */
+  public get subcategories(): Set<Id64String> {
+    return this.#selection.subcategories;
+  }
+
+  /** Get the active selection as a collection of geometric element, model and subcategory ids.
+   * @note Do not the sets in returned collection directly. Instead, use methods like [[SelectionSet.add]].
```
Comment (bbastings) [inline core/frontend/src/SelectionSet.ts:351]: nitpick...missing "modify" in note...

**Review by bbastings — APPROVED** [review]

**Review by saskliutas — APPROVED** [review]

File: core/frontend/src/SelectionSet.ts (hunk 2 above)
Comment (grigasp) [reply core/frontend/src/SelectionSet.ts]: fixed

File: core/frontend/src/test/SelectionSet.test.ts [hunk 3]
```diff
@@ -0,0 +1,1191 @@
...
+      expect(onChangedSpy).lastCalledWith({
+        type: SelectionSetEventType.Add,
+        set: ss,
+        added: expect.toBeId64Arg("0x1"),
+        additions: {
+          elements: expect.toBeId64Arg("0x1"),
+          models: expect.toBeId64Arg(["0x2"]),
+          subcategories: expect.toBeId64Arg(["0x3"]),
+        },
+      });
```
[the remark this answers is not in the document]
Comment (grigasp) [reply core/frontend/src/test/SelectionSet.test.ts:65]: These are two different things and they won't necessarily match. 

The event is raised with changes (e.g. if you call `add` with `["0x1", "0x2"]` and `0x1` is already in the selection set, then the event will only have `0x2` in `additions.elements`.

There are a number of tests where event arguments don't match selection set content. Take `adds element ids to selection set` as an example - we add `0x123` element id, the event is raised with it, but the selection set also contains a model, a subcategory and a different element.

**Review by JonasDov — APPROVED** [review]

**Pull request merged**